Unity Package Manager fails on Windows with "undefined path" error

## Bug Description

When running the Unity MCP server on Windows, Unity fails to start with a Package Manager error: "The 'path' argument must be of type string. Received undefined". This prevents the MCP server from functioning properly on Windows systems.

## Steps to Reproduce

1. Install and configure the Unity MCP server on Windows according to the steps in README.md
2. Launch Unity through your AI agent's MCP integration
3. Unity attempts to start but fails with Package Manager error

## Expected Behavior

Unity should start successfully and the MCP server should function normally, as it does on macOS and Linux.

## Actual Behavior

Unity displays error dialog:
```
Failed to resolve packages: The "path" argument must be of type string. Received undefined. No packages loaded.

Click on Retry to relaunch Unity and reopen your project.
Click on Continue to launch Unity. Some or all packages may not be imported which may cause compilation errors.
```

Unity's Package Manager diagnostics show:
```
[ERROR] The "path" argument must be of type string. Received undefined
```

## Environment

- **OS**: Windows 11 (10.0.26100) 64bit Professional
- **Unity Version**: 6000.1.10f1
- **Node.js Version**: [Current LTS]
- **Package Version**: Latest from npm (@nurture-tech/unity-mcp-runner)
- **Agent**: Augment Code (VS Code)
## Root Cause Analysis

The issue occurs because:

1. **Command line arguments bug**: The original code passes all process arguments to Unity (`process.argv.slice(1)`), including the script path and MCP runner arguments, which Unity doesn't understand.

2. **Missing Windows environment variables**: Unity's Package Manager requires specific Windows environment variables (`PROGRAMDATA`, `ALLUSERSPROFILE`, `SYSTEMROOT`, `LOCALAPPDATA`, `USERPROFILE`) to resolve configuration paths. When these are undefined, Unity's `getLocalConfigFolder()` and `getCloudConfigCacheFolder()` functions fail with "path argument must be of type string" errors.

--- a/packages/node-runner/src/index.ts
+++ b/packages/node-runner/src/index.ts
@@ -90,7 +90,30 @@ const packageJson = JSON.parse(await fs.readFile(packageJsonPath, "utf8"));
 packageJson.dependencies["is.nurture.mcp"] = packageUrl;
 await fs.writeFile(packageJsonPath, JSON.stringify(packageJson, null, 2));
 
-const proc = spawn(unityPath, [...process.argv.slice(1), "-mcp", "-logFile", "-"]);
+// Only pass Unity-specific arguments, not the MCP runner arguments
+const unityArgs = ["-projectPath", args.projectPath, "-mcp", "-logFile", "-"];
+
+// Check if there are additional Unity arguments after "--" separator
+const separatorIndex = process.argv.indexOf("--");
+if (separatorIndex !== -1) {
+  unityArgs.push(...process.argv.slice(separatorIndex + 1));
+}
+
+// Set up environment variables for Unity Package Manager
+const env = {
+  ...process.env,
+};
+
+if (process.platform === "win32") {
+  // Fix for Windows: Unity Package Manager needs these environment variables
+  const userProfile = process.env.USERPROFILE || "C:\\Users\\Default";
+  env.PROGRAMDATA = process.env.PROGRAMDATA || "C:\\ProgramData";
+  env.ALLUSERSPROFILE = process.env.ALLUSERSPROFILE || "C:\\ProgramData";
+  env.SYSTEMROOT = process.env.SYSTEMROOT || "C:\\Windows";
+  env.LOCALAPPDATA = process.env.LOCALAPPDATA || path.join(userProfile, "AppData", "Local");
+}
+
+const proc = spawn(unityPath, unityArgs, { env });
 
 try {
   const code = await new Promise<number | null>((resolve, reject) => {

